it/src/main/java/com/google/cloud/teleport/it/dataflow/DataflowJobUtils.java

```diff
@@ -16,13 +16,42 @@
 package com.google.cloud.teleport.it.dataflow;
 
 import com.google.common.base.CaseFormat;
+import java.io.IOException;
 import java.time.Instant;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.beam.sdk.PipelineResult;
 
 /** Utilities to make working with Dataflow easier. */
-public final class DataflowUtils {
-  private DataflowUtils() {}
+public class DataflowJobUtils {
+
+  public static boolean waitUntilState(
```

Comment:
DataflowOperator requires a DataflowTemplateClient. It does not support jobs that are not launched from templates. We could delegate the DataflowOperator implementation to be based on DataflowJobUtils instead, which is the more general method - but this looks like it would require a larger refactor of Dataflow Operator functionality.

What do you prefer?